pkg/kubelet/pleg/generic.go

```diff
@@ -251,10 +251,34 @@ func (g *GenericPLEG) Relist() {
 	for pid := range g.podRecords {
 		oldPod := g.podRecords.getOld(pid)
 		pod := g.podRecords.getCurrent(pid)
+
+		var cachePodStatus *kubecontainer.PodStatus
+		var podStatus *kubecontainer.PodStatus
+
+		// get cachePodStatus and runtimePodStatus to help distinguish the resized pod.
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			if !g.cacheEnabled() {
+				klog.ErrorS(err, "GenericPLEG: Cache is not enabled")
+				return
+			}
+			cachePodStatus, err = g.cache.Get(pid)
+			if err != nil {
+				klog.ErrorS(err, "GenericPLEG: Unable to retrieve pods", "pid", pid)
+				return
+			}
+			if pod != nil {
+				podStatus, err = g.runtime.GetPodStatus(ctx, pod.ID, pod.Name, pod.Namespace)
```

Comment:
Do we have to call it for all pods? Wouldn't it be enough to get a pod status from the runtime only when the pod is being resized (`InProgress`)?